crates/scroll/chainspec/src/lib.rs

```diff
@@ -0,0 +1,592 @@
+            (EthereumHardfork::Constantinople.boxed(), genesis.config.constantinople_block),
+            (EthereumHardfork::Petersburg.boxed(), genesis.config.petersburg_block),
+            (EthereumHardfork::Istanbul.boxed(), genesis.config.istanbul_block),
+            (EthereumHardfork::Berlin.boxed(), genesis.config.berlin_block),
+            (EthereumHardfork::London.boxed(), genesis.config.london_block),
+            (ScrollHardfork::Archimedes.boxed(), hard_fork_info.archimedes_block),
+            (ScrollHardfork::Bernoulli.boxed(), hard_fork_info.bernoulli_block),
+            (ScrollHardfork::Curie.boxed(), hard_fork_info.curie_block),
+        ];
+        let mut block_hardforks = hardfork_opts
+            .into_iter()
+            .filter_map(|(hardfork, opt)| opt.map(|block| (hardfork, ForkCondition::Block(block))))
+            .collect::<Vec<_>>();
+
+        // Time-based hardforks
+        let time_hardfork_opts = [
+            (EthereumHardfork::Shanghai.boxed(), genesis.config.shanghai_time),
+            (ScrollHardfork::Darwin.boxed(), hard_fork_info.darwin_time),
+            (ScrollHardfork::DarwinV2.boxed(), hard_fork_info.darwin_v2_time),
+        ];
+
+        let mut time_hardforks = time_hardfork_opts
+            .into_iter()
+            .filter_map(|(hardfork, opt)| {
+                opt.map(|time| (hardfork, ForkCondition::Timestamp(time)))
+            })
+            .collect::<Vec<_>>();
+
+        block_hardforks.append(&mut time_hardforks);
+
+        // Ordered Hardforks
+        let mainnet_hardforks = ScrollHardfork::scroll_mainnet();
+        let mainnet_order = mainnet_hardforks.forks_iter();
+
+        let mut ordered_hardforks = Vec::with_capacity(block_hardforks.len());
+        for (hardfork, _) in mainnet_order {
+            if let Some(pos) = block_hardforks.iter().position(|(e, _)| **e == *hardfork) {
+                ordered_hardforks.push(block_hardforks.remove(pos));
+            }
+        }
+
+        // append the remaining unknown hardforks to ensure we don't filter any out
+        ordered_hardforks.append(&mut block_hardforks);
+
+        Self {
+            inner: ChainSpec {
+                chain: genesis.config.chain_id.into(),
+                genesis,
+                hardforks: ChainHardforks::new(ordered_hardforks),
+                ..Default::default()
+            },
+        }
+    }
+}
+
+#[derive(Default, Debug)]
+struct ScrollConfigInfo {
+    scroll_chain_info: ScrollChainInfo,
+}
+
+impl ScrollConfigInfo {
+    fn extract_from(genesis: &Genesis) -> Self {
+        Self {
+            scroll_chain_info: ScrollChainInfo::extract_from(&genesis.config.extra_fields)
+                .expect("extract scroll extra fields failed"),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::*;
+    use alloy_genesis::{ChainConfig, Genesis};
+    use reth_ethereum_forks::EthereumHardfork;
+    use reth_scroll_forks::ScrollHardfork;
+
+    // #[test]
```

Comment:
nit: can we maybe uncomment and add an `#[ignore("waiting on https://github.com/scroll-tech/reth/pull/36")]` on each test?

Reply:
modified [1a69d3a](https://github.com/scroll-tech/reth/pull/56/commits/1a69d3a8e583f63bc05efbc2055f9b510b4490d7)